src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Debug/LaunchSettingsProvider.cs

```diff
@@ -44,28 +42,28 @@ internal class LaunchSettingsProvider : OnceInitializedOnceDisposed, ILaunchSett
         public const string ErrorProfileCommandName = "ErrorProfile";
 
         private readonly UnconfiguredProject _project;
-        private readonly IActiveConfiguredValue<IAppDesignerFolderSpecialFileProvider> _appDesignerSpecialFileProvider;
+        private readonly IActiveConfiguredValue<IAppDesignerFolderSpecialFileProvider?> _appDesignerSpecialFileProvider;
         private readonly IProjectFaultHandlerService _projectFaultHandler;
         private readonly AsyncLazy<string> _launchSettingsFilePath;
         private readonly IUnconfiguredProjectServices _projectServices;
         private readonly IUnconfiguredProjectCommonServices _commonProjectServices;
-        private readonly IActiveConfiguredProjectSubscriptionService _projectSubscriptionService;
+        private readonly IActiveConfiguredProjectSubscriptionService? _projectSubscriptionService;
         private readonly IFileSystem _fileSystem;
         private readonly TaskCompletionSource<bool> _firstSnapshotCompletionSource = new TaskCompletionSource<bool>();
-        private SequentialTaskExecutor _sequentialTaskQueue = new SequentialTaskExecutor();
-        private IReceivableSourceBlock<ILaunchSettings> _changedSourceBlock;
-        private IBroadcastBlock<ILaunchSettings> _broadcastBlock;
-        private ILaunchSettings _currentSnapshot;
-        private IDisposable _projectRuleSubscriptionLink;
+        private readonly SequentialTaskExecutor _sequentialTaskQueue = new SequentialTaskExecutor();
+        private IReceivableSourceBlock<ILaunchSettings>? _changedSourceBlock;
+        private IBroadcastBlock<ILaunchSettings>? _broadcastBlock;
+        private ILaunchSettings? _currentSnapshot;
+        private IDisposable? _projectRuleSubscriptionLink;
 
         [ImportingConstructor]
         public LaunchSettingsProvider(
             UnconfiguredProject project,
             IUnconfiguredProjectServices projectServices,
             IFileSystem fileSystem,
             IUnconfiguredProjectCommonServices commonProjectServices,
-            IActiveConfiguredProjectSubscriptionService projectSubscriptionService,
-            IActiveConfiguredValue<IAppDesignerFolderSpecialFileProvider> appDesignerSpecialFileProvider,
+            IActiveConfiguredProjectSubscriptionService? projectSubscriptionService,
```

Comment:
This will never be null.

Reply:
`null` is being passed in tests, and the implementation explicitly tests for `null` to avoid setting up dataflow subscriptions.